ee/tabby-webserver/src/service/auth.rs

```diff
@@ -47,27 +54,28 @@ pub fn create(
     license: Arc<dyn LicenseService>,
     setting: Arc<dyn SettingService>,
 ) -> impl AuthenticationService {
-    let mut impersonate = None;
+    create_impl(db, mail, license, setting)
+}
+
+fn create_impl(
+    db: DbConn,
+    mail: Arc<dyn EmailService>,
+    license: Arc<dyn LicenseService>,
+    setting: Arc<dyn SettingService>,
+) -> AuthenticationServiceImpl {
+    let mut impersonate_user = None;
     if let Ok(value) = std::env::var("TABBY_OWNER_IMPERSONATE_OVERRIDE") {
         let words: Vec<&str> = value.split(':').collect();
         if words.len() == 2 {
-            impersonate = Some((words[0].to_string(), words[1].to_string()));
+            let password_encrypted = password_hash(words[1]).expect("can not encrypt passworkd");
+            impersonate_user = Some(ImpersonateUserCredential {
+                // impersonate user is owner use, so use 1 as id.
```

Comment:
Please pay more attention to english writing, e.g try optimize it with answer engine:

https://jam.dev/c/57400152-1b1a-4d6f-bb02-036c45dae34a